src/main/java/com/flowpowered/nbt/regionfile/Chunk.java

```diff
@@ -0,0 +1,298 @@
+ * 
+ * @author piegames
+ */
+public class Chunk {
+	/** The x coordinate of the chunk relative to its RegionFile's origin */
+	public final int			x;
+	/** The z coordinate of the chunk relative to its RegionFile's origin */
+	public final int			z;
+	/**
+	 * The point in time when this chunk last got written. Equal to {@code (int) (System.currentTimeMillis() / 1000L)}
+	 */
+	public final int			timestamp;
+
+	protected final ByteBuffer	data;
+
+	/**
+	 * Create a Chunk object by setting its data directly through a buffer.
+	 * 
+	 * @param x
+	 *            The x coordinate of the chunk within its region file. Must be in the range of [0..32) because region files are always 32*32
+	 *            chunks large.
+	 * @param z
+	 *            The z coordinate of the chunk within its region file. Must be in the range of [0..32) because region files are always 32*32
+	 *            chunks large.
+	 * @param data
+	 *            The data of the chunk
+	 * @throws IllegalArgumentException
+	 *             if the coordinates are out of bounds or the data does not have a size multiple of 4096
+	 * @author piegames
+	 */
+	public Chunk(int x, int z, int timestamp, ByteBuffer data) {
+		if (x < 0 || z < 0 || x >= 32 || z >= 32)
+			throw new IllegalArgumentException("Coordinates must be in range [0..32), but were x=" + x + ", z=" + z + ")");
+		this.x = x;
+		this.z = z;
+		this.timestamp = timestamp;
+
+		if ((data.capacity() & 4095) != 0)
+			throw new IllegalArgumentException("Data buffer size must be multiple of 4096, but is " + data.capacity());
+		this.data = Objects.requireNonNull(data);
+	}
+
+	/**
+	 * Create a Chunk object by reading specified data from a region file (*.mca, *.mcr).
+	 * 
+	 * @param x
+	 *            The x coordinate of the chunk within its region file. Must be in the range of [0..32) because region files are always 32*32
+	 *            chunks large.
+	 * @param z
+	 *            The z coordinate of the chunk within its region file. Must be in the range of [0..32) because region files are always 32*32
+	 *            chunks large.
+	 * @param raf
+	 *            The file channel to the region file from which to load the data
+	 * @param start
+	 *            The number of the 4096 byte sector where the chunk is located in the file. Don't forget that the first five bytes are used to
+	 *            store the size and compression of the chunk. The position of the first byte of NBT data is thus {@code start*4096 + 5}.
+	 * @param length
+	 *            The amount of 4096 byte sectors to load. It should be large enough to contain all NBT data in that chunk or it will be
+	 *            corrupted.
+	 * @throws IllegalArgumentException
+	 *             if the coordinates are out of bounds
+	 * @author piegames
+	 */
+	public Chunk(int x, int z, int timestamp, FileChannel raf, int start, int length) throws IOException {
```

Comment:
This constructor should not be public, but instead be package private. (or maybe not even exist, but that is up to you)

Reply:
Will do.